### PR TITLE
Rename Install CRD to KnativeServing, fixes #17

### LINKED ISSUE #17: Consider renaming CR kind to "KnativeServing"
I believe, "Install" as a name for the CR is too generic. Currently, its also used by eventing (and Tekton pipeline install operator).

I suggest to rename it to "KnativeServing" and "KnativeEventing" for the corresponding eventing operator.

Why ?

* `kubectl get install` would need to be always fully qualified with the API group if more than one "install" CR is present (`kubectl get installs.serving.knative.de`)
* "install" is too generic and is more a verb than a noun like CRs typically are (therefore that singular/plural thing in the CRD definition)
* Other products install operators like for Prometheus or Etcd also call their CR after the product (i.e. "Prometheus" and "Etcd", respectively)
* Its easier to understand a `kind: KnativeServing` alone then checking the `apiVersion: serving.knative.dev/v1alpha1` always together with the `kind`.

If we want to change that name, we should do it ASAP, i.e. before going to upstream with it.

### PR DESCRIPTION
**Pull request by jcrossley3**

### CODE REVIEW

File: deploy/olm-catalog/knative-serving-operator/0.6.0/serving_v1alpha1_knativeserving_crd.yaml [hunk 1]
```diff
@@ -1,14 +1,14 @@
...
-    kind: Install
-    listKind: InstallList
-    plural: installs
-    singular: install
+    kind: KnativeServing
+    listKind: KnativeServingList
+    plural: knativeservings
+    singular: knativeserving
     shortNames:
     - ksi
```
Comment (matzew) [inline deploy/olm-catalog/knative-serving-operator/0.6.0/serving_v1alpha1_knativeserving_crd.yaml]: @jcrossley3 should that `ksi` be `ks` only  ?

**Comment by matzew** [comment]
I think this needs to be updated too:

https://github.com/openshift/knative-serving/blob/release-v0.6.0/openshift/olm/knative-serving.catalogsource.yaml#L299

File: deploy/olm-catalog/knative-serving-operator/0.6.0/serving_v1alpha1_knativeserving_crd.yaml (hunk 1 above)
Comment (jcrossley3) [reply deploy/olm-catalog/knative-serving-operator/0.6.0/serving_v1alpha1_knativeserving_crd.yaml]: Good call!

Comment (matzew) [reply deploy/olm-catalog/knative-serving-operator/0.6.0/serving_v1alpha1_knativeserving_crd.yaml]: OK, I will update my PR w/ `ke` .. fwiw

**Comment by jcrossley3** [comment]
> so, we keep `knative.dev`, instead of `not-knative.dev`?

For the moment, yes, per @bbrowning, but that's a fairly trivial change compared to this one, so we can do that any time.

**Comment by jcrossley3** [comment]
> I think this needs to be updated too:
> 
> https://github.com/openshift/knative-serving/blob/release-v0.6.0/openshift/olm/knative-serving.catalogsource.yaml#L299

Agree. I'll regenerate the catalog and stick it up there after this is merged.

**Pull request merged**